compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -201,18 +202,24 @@ fn emit_orphan_check_error<'tcx>(
     tcx: TyCtxt<'tcx>,
     sp: Span,
     trait_span: Span,
+    self_ty: Ty<'tcx>,
     self_ty_span: Span,
     generics: &hir::Generics<'tcx>,
     err: traits::OrphanCheckErr<'tcx>,
 ) -> Result<!, ErrorGuaranteed> {
     Err(match err {
         traits::OrphanCheckErr::NonLocalInputType(tys) => {
+            let msg = match self_ty.kind() {
+                ty::Adt(..) => "can be implemented for types defined outside of the crate",
+                ty::Param(_) => "can have blanket implementations defined in this trait",
```

Comment:
Can you add a test case that emits this?

Reply:
The following lines seem to correspond to errors for `ty::Param`, so I'll remove this line.

https://github.com/rust-lang/rust/blob/10baaa6ed243c3e026491b1068c44356eaf24069/compiler/rustc_typeck/src/coherence/orphan.rs#L311-L314